datadog/api/aws_integration.py

```diff
@@ -0,0 +1,203 @@
+        :type host_tags: list of strings
+
+        :param account_specific_namespace_rules: An object (in the form \
+        {"namespace1":true/false, "namespace2":true/false}) that enables \
+        or disables metric collection for specific AWS namespaces for this \
+        AWS account only. A list of namespaces can be found at the \
+        /v1/integration/aws/available_namespace_rules endpoint.
+        :type account_specific_namespace_rules: dictionary
+
+        :returns: Dictionary representing the API's JSON response
+
+        >>> account_id = "<AWS_ACCOUNT_ID>"
+        >>> access_key_id = "<AWS_ACCESS_KEY_ID>"
+        >>> role_name = "DatadogAwsRole"
+        >>> filter_tags = ["<KEY>:<VALUE>"]
+        >>> host_tags = ["<KEY>:<VALUE>"]
+        >>> account_specific_namespace_rules = {"namespace1":true/false, "namespace2":true/false}
+
+        >>> api.Aws.create(account_id=account_id, role_name=role_name, \
+        filter_tags=filter_tags,host_tags=host_tags,\
+        account_specific_namespace_rules=account_specific_namespace_rules)
+        """
+        return super(Aws, cls).create(id=id, **params)
+
+    @classmethod
+    def update(cls, id=_resource_id, **body):
+        """
+        Update an AWS integration config.
+
+        :param account_id: Your AWS Account ID without dashes. \
+        Consult the Datadog AWS integration to learn more about \
+        your AWS account ID.
+        :type account_id: string
+
+        :param existing_account_id: Your existing AWS Account ID without dashes. \
+        Consult the Datadog AWS integration to learn more about \
+        your AWS account ID. This is the account to be updated.
+        :type account_id: string
+
+        :param access_key_id: If your AWS account is a GovCloud \
+        or China account, enter the corresponding Access Key ID.
+        :type access_key_id: string
+
+        :param existing_access_key_id: If your AWS account is a GovCloud \
+        or China account, enter the existing Access Key ID to be changed.
+        :type access_key_id: string
+
+        :param secret_access_key: If your AWS account is a GovCloud \
+        or China account, enter the corresponding Secret Access Key.
+        :type access_key_id: string
+
+        :param existing_secret_access_key: If your AWS account is a GovCloud \
+        or China account, enter the existing key to be changed.
+        :type access_key_id: string
+
+        :param role_name: Your Datadog role delegation name. \
+        For more information about you AWS account Role name, \
+        see the Datadog AWS integration configuration info.
+        :type role_name: string
+
+        :param existing_role_name: Your existing Datadog role delegation name. \
+        For more information about you AWS account Role name, \
+        see the Datadog AWS integration configuration info. \
+        This is the role_name to be updated.
+        :type role_name: string
```

Comment:
This was done because the endpoint in dogweb expects query parameters of account_id, role_name, etc. AND a JSON body containing account_id, role_name, etc. Those fields being sent as query params being the ones representing the existing fields. 

I'm unsure on the best way to manipulate the body object being passed in with the `new` as they need to remain unchanged in the body object being passed in to line up with expected fields. 

i.e. if we just call the existing_account_id, account_id, we still need to get this out of the body object to be passed in as query params, and we would also then need to introduce logic that will manipulate the `new_account_id` field in the body to change into `account_id`. This would also require additional logic. i.e. my early draft of how this would need to look is as follows:

```
       if body.get('account_id') and body.get('role_name'):
            params['account_id'] = body.pop('account_id')
            params['role_name'] = body.pop('role_name')
            if body.get('new_account_id'):
                body['account_id'] = body.pop('new_account_id')
            if body.get('new_role_name'):
                body['role_name'] = body.pop('new_role_name')
        if body.get('access_key_id'):
            params['access_key_id'] = body.pop('access_key_id')
            params['secret_access_key'] = body.pop('secret_access_key')
            if body.get('new_access_key_id'):
                body['access_key_id'] = body.pop('new_access_key_id')
            if body.get('new_secret_access_key'):
                body['secret_access_key'] = body.pop('new_secret_access_key')
```

That said, I'm having some trouble getting this update endpoint working now with the access_key and secret_key setup. I had that working before I had thought, so revisiting this entirely to make sure it works in all cases.

Reply:
Yeah this sound good to me. I don’t mind having a bit more logic if it means the user experience will be better and more consistent. The update endpoint for azure works like that, so better to have things consistent here.

Reply:
This has been updated now to work the way you'd mentioned. I agree it's easier to use. That said, this also surfaced some potential bugs in the AWS Update endpoint in dogweb. I'm looking into that further tomorrow and will potentially be changing this again.

Reply:
This is now working properly with the updated method you'd suggested. I've fixed the bug that was surfaced by this work and should now be good to go!